src/components/calcite-action/calcite-action.tsx

```diff
@@ -120,9 +125,12 @@ export class CalciteAction {
   }
 
   renderIconContainer(): VNode {
-    const { loading } = this;
+    const { el, loading, icon } = this;
 
-    const calciteLoaderNode = <calcite-loader is-active inline></calcite-loader>;
+    const calciteLoaderNode = loading ? <calcite-loader is-active inline></calcite-loader> : null;
+    const calciteIconNode = icon ? <calcite-icon icon={icon} scale="s"></calcite-icon> : null;
```

Comment:
I suggest having the the Icon's `scale` be related to the Action's `scale`. A direct relationship ain't great, but this seems to work nicely.

```
const iconScale = scale !== "l" ? "s" : "m";
```
The `scale` here being from calcite-action.

Reply:
Basically, the icon is small unless action is large, then icon is medium.